miio/tests/test_devicestatus.py

```diff
@@ -247,6 +250,24 @@ def level(self) -> TestEnum:
     setter.assert_called_with(TestEnum.Second)
 
 
+@pytest.mark.parametrize("getter_name", ["actions", "settings", "sensors"])
+def test_cached_descriptors(getter_name, mocker):
+    d = Device("127.0.0.1", "68ffffffffffffffffffffffffffffff")
+    getter = getattr(d, getter_name)
+    initialize_descriptors = mocker.patch(
+        "miio.Device._initialize_descriptors",
+        side_effect=Device._initialize_descriptors,
+        autospec=True,
+    )
+    mocker.patch("miio.Device.status")
+    mocker.patch("miio.Device._sensor_descriptors_from_status", return_value={})
+    mocker.patch("miio.Device._setting_descriptors_from_status", return_value={})
+    mocker.patch("miio.Device._action_descriptors", return_value={})
```

Comment:
These should not be needed as `_initialize_descriptors` is mocked above, and that's the only call-site for these.

Reply:
No the mock of _initialize_descriptors has as side_effect itself which makes sure the code of _initialize_descriptors is actually executed. The mock is just used to count the number of calls, but the function is not overwritten.

We need _initialize_descriptors to be executed, otherwise the self._sensors etc. are not set and therefore nothing is cached...

Reply:
Ah, I see. Instead of defining a side effect, it's better to use a spy there: https://pytest-mock.readthedocs.io/en/latest/usage.html#spy

Reply:
I just noticed this is in the wrong file, the correct place would be `test_device.py`. Otherwise this looks good to go, I'll merge as soon as tests are passing.

Reply:
Thanks for the tip to use spy!
Learning each day :)

I have moved the test to the test_device.py file.